pkg/models/openpitrix/repos.go

```diff
@@ -298,8 +298,6 @@ func (c *repoOperator) ListRepos(conditions *params.Conditions, orderBy string,
 	}
 
 	totalCount := len(repos)
-	start, end := (&query.Pagination{Limit: limit, Offset: offset}).GetValidPagination(totalCount)
-	repos = repos[start:end]
 	items := make([]interface{}, 0, len(repos))
 	for i, j := offset, 0; i < len(repos) && j < limit; i, j = i+1, j+1 {
```

Comment:
I think we should not to remove those line, but should fix the for loop.
when we want to get the second page, the `start` is 10, `end` is 11, `len(repos)` is 1 and `i` is 10, so if `i < len(repos)` is false

Reply:
How about this?
```
	start, end := (&query.Pagination{Limit: limit, Offset: offset}).GetValidPagination(totalCount)
	repos = repos[start:end]
	items := make([]interface{}, 0, len(repos))
	for i := range repos {
		items = append(items, convertRepo(repos[i]))
	}
```

Reply:
more better

Reply:
> How about this?
> 
> ```
> 	start, end := (&query.Pagination{Limit: limit, Offset: offset}).GetValidPagination(totalCount)
> 	repos = repos[start:end]
> 	items := make([]interface{}, 0, len(repos))
> 	for i := range repos {
> 		items = append(items, convertRepo(repos[i]))
> 	}
> ```

Well, this way.